### DISCUSSION TITLE
What is "webpackEmptyAsyncContext" and why does it return Error?

### DISCUSSION THREAD

**Discussion by serdaryesilmurat** [webpack/webpack#16732] [Q&A]
(First of all, I am using CRA 4.)

In my code I have a dynamic import statement that loads a JS bundle file from a remote path.

> import("http://localhost:10000/index.modern.js");

When webpack compiles the code it replaces "import" statement with 

>  __webpack_require__("./node_modules/my-module/dist lazy recursive")();

And this call refers to the webpackEmptyAsyncContext function generated by webpack:

```
/*! no static exports found */
/***/ (function(module, exports) {

function webpackEmptyAsyncContext(req) {
	// Here Promise.resolve().then() is used instead of new Promise() to prevent
	// uncaught exception popping up in devtools
	return Promise.resolve().then(function() {
		var e = new Error("Cannot find module '" + req + "'");
		e.code = 'MODULE_NOT_FOUND';
		throw e;
	});
}
webpackEmptyAsyncContext.keys = function() { return []; };
webpackEmptyAsyncContext.resolve = webpackEmptyAsyncContext;
module.exports = webpackEmptyAsyncContext;
webpackEmptyAsyncContext.id = "./node_modules/my-module/dist lazy recursive";
```

What am I supposed to do to make webpack to use "import()" instead of webpackEmptyAsyncContext in the generated bundle.

**Comment by alexander-akait** [comment]
Looks like you have `require("my-package/" + name)` code and webpack can't found any files which can handled, try to debug your code and find a place there is you have such code
